src/views/pages/Diagnostics/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import Node from '../../../services/Node';
+
+function Diagnostics() {
+
+    const [data, setData] = useState({ pending: true, diagnostics: null });
+
+    useEffect(() => {
+        setData({ pending: true, diagnostics: {} });
+
+        const fetchData = async () => {
+            const node = new Node();
+            const data = await node.getDiagnosticsData();
+            return data;
+        }
+        fetchData().then((fetchedData) => {
+            setData({
+                pending: false,
+                diagnostics: fetchedData
+            })
+        }).catch(console.error);
```

Comment:
instead of using `pending` -- you could define the initial state as "undefined" and while it remains undefined, that tells you, conceptually, that `pending = false`

In the react code, you'd conditionally display the view like thisl `<div>{data && <p>my view<p>}</div>`